pyproject.toml

```
@@ -0,0 +1,47 @@
+[tool.poetry]
+name = "upstash-semantic-cache"
+version = "1.0.0"
+description = "Semantic Caching Algorithm for Upstash Vector"
+license = "MIT"
+authors = ["Upstash <support@upstash.com>"]
+maintainers = ["Upstash <support@upstash.com>"]
+readme = "README.md"
+repository = "https://github.com/upstash/semantic-cache-py"
+keywords = ["Upstash", "Upstash Vector", "Semantic Cache"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Database",
+    "Topic :: Software Development :: Libraries",
+]
+
+packages = [{ include = "upstash_semantic_cache" }]
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+langchain_core = "0.2.0"
+upstash_vector = "0.5.0"
```

```suggestion
upstash_vector = ">=0.5.0, <1"
```

so that we don't have to update this dependency when we release a new version of the upstash_vector. I think the APIs we use in this code will always work from the upstash vector